### PR TITLE
Fix cookies not being forwarded in graphql request

### PR DESCRIPTION
**Pull request by lucasecdb**

### CODE REVIEW

File: node/middlewares/schema.ts [hunk 1]
```diff
@@ -34,13 +37,18 @@ const executor = (app: string): AsyncExecutor => async ({
...
         Authorization: authToken,
+        cookie: headers.cookie,
         'x-vtex-locale': 'en-US',
         'x-vtex-tenant': 'en-US',
       },
       body: JSON.stringify({ query, variables }),
     }
   )
 
+  if (shouldForwardCookie) {
```
Comment (tlgimenes) [inline node/middlewares/schema.ts:48]: if this is always true, why adding this if ?

Comment (lucasecdb) [reply node/middlewares/schema.ts:48]: it's not always true, for the introspection request we don't want to forward the cookies (here https://github.com/vtex-apps/graphql-gateway/blob/fix/cookies/node/middlewares/schema.ts#L61)

**Comment by lucasecdb** [comment]
> One thing I couldn't understand is if we are varying according to the cookie, and if `search` queries will also vary with the cookie.

what do you mean? I don't think I understood 😅

> Cookies are an antipattern in graphql and all data should be sent via querie's variables. This way we only vary whenever necessary and we don't require magic cdn configs/features

but all of our queries in `checkout-graphql` already consider that you have the `checkout.vtex.com` cookie in order for the API to give us the correct orderForm, and the `store-graphql` was also recently changed to deprecate the `orderFormId` argument in favor of the cookie.

File: node/middlewares/schema.ts (hunk 1 above)
Comment (tlgimenes) [reply node/middlewares/schema.ts:48]: Cool, I haven't seen that one. Anyways, the introspection is done only once, so it shouldn't be a problem

**Comment by tlgimenes** [comment]
> what do you mean? I don't think I understood 😅

This gateway is used for both checkout and search. Imagine the scenario where a cookied user is browsing the store. The question is: will this user hit the same cache as a non cookied user that browsered the store before ?

**Comment by lucasecdb** [comment]
we've decided to not use cookies in the GraphQL queries/mutations anymore, so I'm going to close this

**Pull request closed without merging**